packages/flutter_tools/lib/src/cache.dart

```diff
@@ -356,6 +356,30 @@ class Cache {
     }
   }
 
+  String get devToolsVersion {
+    if (_devToolsVersion == null) {
+      final Directory devToolsDir = getCacheDir('dart-sdk/bin/resources/devtools');
+      if (!devToolsDir.existsSync()) {
+        throwToolExit('Could not determine DevTools version.');
+      }
+      final File versionFile = _fileSystem.file('${devToolsDir.path}/version.json');
+      if (!versionFile.existsSync()) {
+        throwToolExit('Could not determine DevTools version.');
+      }
+      final dynamic data = jsonDecode(versionFile.readAsStringSync());
+      if (data is! Map<String, Object>) {
+        throwToolExit('Could not determine DevTools version.');
```

Comment:
and this should be a different exception, making it clear we found the file but failed to parse it

Reply:
done